### ISSUE TITLE
Wifi manager — Unable to connect WPA and WPA2 wifi, there aren't list as secure

### ISSUE THREAD

**Issue by Natim** [ideascube/ideascube#231]
The wifi manager adds a lock on WEP protected wifi but not on WPA/WPA2 networks.
Thus it is not possible to enter a password.

**Comment by bochecha** [comment]
I think there are two issues here.

First of all, there is the question of adding a configuration UI in the history view. I guess we could do that, yes. However, to me this is completely orthogonal to your connexion problems (i.e your problem needs to be fixed whether we add some configuration UI or not). Can you open a new issue to track this one separately?

And then, let's focus this issue on your connection problem.

For WPA/WPA2 networks, Ideascube should definitely display the lock to indicate that they are protected, and it should also ask you for a password when you click on the network.

If that's not what you see, then it's a bug which we have to fix.

I'm a bit surprised though, because it works for me here. :confused: 

So, in your ideascube virtualenv, can you run the following script?

```python
from operator import attrgetter
from NetworkManager import NetworkManager, NM_DEVICE_TYPE_WIFI


for device in NetworkManager.Devices:
    if device.Managed and device.DeviceType != NM_DEVICE_TYPE_WIFI:
        continue

    aps = device.SpecificDevice().GetAllAccessPoints()
    for ap in sorted(aps, key=attrgetter('Ssid')):
        print('SSID: "%s", WPA Flags: %s' % (ap.Ssid, ap.WpaFlags))
```

Then, can you attach the output here, as well as the SSID of the network you're trying to connect to?

**Comment by bochecha** [comment]
Sorry, that script I gave you in my previous comment was wrong.

Now that I have had a chance to try it out, I realize this is what I need you to try:

```python
from operator import attrgetter
from NetworkManager import NetworkManager, NM_DEVICE_TYPE_WIFI


for device in NetworkManager.Devices:
    if not device.Managed or device.DeviceType != NM_DEVICE_TYPE_WIFI:
        continue

    aps = device.SpecificDevice().GetAllAccessPoints()
    for ap in sorted(aps, key=attrgetter('Ssid')):
        print('SSID: "%s", WPA Flags: %s' % (ap.Ssid, ap.WpaFlags))
```

Apologies for that, I had written the previous one from looking at the ideascube sources, without a mean to test it.

**Comment by Natim** [comment]
I will let @fheslouin try it on the Koombook.

I have created the issue here: https://github.com/ideascube/ideascube/issues/232

**Comment by fheslouin** [comment]
I manage to reproduce the issue with the KoomBook. @bochecha I fellowed wrote the python command in the ideascube shell and here is what I got : 
```
(2:3)$ ideascube shell
IDEASCUBE_ID=kb_bsfcampus_sen
Importing settings from ideascube.conf.kb_bsfcampus_sen
Python 2.7.9 (default, Mar  1 2015, 13:48:22) 
[GCC 4.9.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from operator import attrgetter
>>> from NetworkManager import NetworkManager, NM_DEVICE_TYPE_WIFI
>>> 
>>> 
>>> for device in NetworkManager.Devices:
...     if not device.Managed or device.DeviceType != NM_DEVICE_TYPE_WIFI:
...         continue
...     aps = device.SpecificDevice().GetAllAccessPoints()
...     for ap in sorted(aps, key=attrgetter('Ssid')):
...         print('SSID: "%s", WPA Flags: %s' % (ap.Ssid, ap.WpaFlags))
... 
SSID: "Bbox-82C87A", WPA Flags: 332
SSID: "Bouygues Telecom Wi-Fi", WPA Flags: 0
SSID: "koombook", WPA Flags: 0
SSID: "splite", WPA Flags: 0
```
From the ideascube web interface ```Bbox-82C87A``` show the lock, but ```splite``` does not show the lock although it is an WPA2 PSK encrypted hotspot. 

I'm gonna try with a fresh install system to verify if I can reproduce again the issue

**Comment by bochecha** [comment]
Thanks for the output @fheslouin.

> From the ideascube web interface Bbox-82C87A show the lock, but splite does not show the lock although it is an WPA2 PSK encrypted hotspot.

This is weird, because:

```
```

Looking at the docs, the `WpaFlags` attribute on the access point is:

https://developer.gnome.org/NetworkManager/unstable/spec.html#org.freedesktop.NetworkManager.AccessPoint
https://developer.gnome.org/NetworkManager/unstable/spec.html#type-NM_802_11_AP_SEC

So I would expect it to not be 0 for a WPA2 PSK hotspot. :confused:

**Comment by bochecha** [comment]
Can you test this new script out?

```python
from operator import attrgetter
from NetworkManager import NetworkManager, NM_DEVICE_TYPE_WIFI


for device in NetworkManager.Devices:
    if not device.Managed or device.DeviceType != NM_DEVICE_TYPE_WIFI:
        continue

    aps = device.SpecificDevice().GetAllAccessPoints()
    for ap in sorted(aps, key=attrgetter('Ssid')):
        print('SSID: "%s", WPA Flags: %s, RsnFlags: %s' % (ap.Ssid, ap.WpaFlags, ap.RsnFlags))
```

**Comment by fheslouin** [comment]
Here is the new output 
```
(2:5)$ ideascube shell
IDEASCUBE_ID=kb_bsfcampus_sen
Importing settings from ideascube.conf.kb_bsfcampus_sen
Python 2.7.9 (default, Mar  1 2015, 13:48:22) 
[GCC 4.9.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from operator import attrgetter
>>> from NetworkManager import NetworkManager, NM_DEVICE_TYPE_WIFI
>>> 
>>> 
>>> for device in NetworkManager.Devices:
...     if not device.Managed or device.DeviceType != NM_DEVICE_TYPE_WIFI:
...         continue
...     aps = device.SpecificDevice().GetAllAccessPoints()
...     for ap in sorted(aps, key=attrgetter('Ssid')):
...         print('SSID: "%s", WPA Flags: %s, RsnFlags: %s' % (ap.Ssid, ap.WpaFlags, ap.RsnFlags))
... 
SSID: "Bbox-82C87A", WPA Flags: 332, RsnFlags: 332
SSID: "Bouygues Telecom Wi-Fi", WPA Flags: 0, RsnFlags: 0
SSID: "koombook", WPA Flags: 0, RsnFlags: 0
SSID: "splite", WPA Flags: 0, RsnFlags: 392
```

**Comment by bochecha** [comment]
Ok, so that confirms what I was told in #nm: `WpaFlags` is not sufficient to understand the security (or lack thereof) of a network, we also must look at `RsnFlags`.

**Comment by bochecha** [comment]
I forgot to mention it in my commit message, but this issue should have been closed by #235.

Don't hesitate to reopen this issue if I missed something and you still can't connect to WPA2 networks. :wink:
